src/scenes/ProgressReports/Detail/ProgressReportCard.tsx

```diff
@@ -16,22 +32,102 @@ interface Props
     | 'resourceType'
     | 'securedFile'
   > {
-  progressReport: Pick<ProgressReportDetailFragment, 'id' | 'reportFile'>;
+  progressReport: Pick<
+    ProgressReportDetailFragment,
+    'id' | 'reportFile' | 'pnpExtractionResult'
+  >;
 }
 
 export const ProgressReportCard = ({ progressReport, ...rest }: Props) => {
+  const file = progressReport.reportFile.value;
+
+  const [reextract, { loading: reextracting }] = useMutation(
+    ReextractPnpProgressDocument,
+    {
+      variables: { reportId: progressReport.id },
+      update: (cache, result) => {
+        cache.modify({
+          id: cache.identify(progressReport),
+          fields: {
+            pnpExtractionResult: () => result.data?.reextractPnpProgress,
+          },
+        });
+      },
+    }
+  );
+
   return (
     <FileActionsContextProvider>
-      <Tooltip title="This holds the progress report PnP file" placement="top">
+      {/*<Tooltip title="This holds the progress report PnP file" placement="top">*/}
+      {/* eslint-disable-next-line react/jsx-no-useless-fragment */}
+      <>
         <DefinedFileCard
           label="PnP File"
           uploadMutationDocument={UploadPeriodicReportFileDocument}
           parentId={progressReport.id}
           resourceType="progress report"
           securedFile={progressReport.reportFile}
+          header={
+            <Stack
+              sx={{
+                mt: -1,
+                flexDirection: 'row',
+                gap: 1,
+                alignItems: 'center',
+              }}
+            >
+              <Typography variant="h3">PnP File</Typography>
+              {file && (
+                <>
+                  <Preview file={file} />
+                  <Feature
+                    flag="pnp-validation"
+                    match={true}
+                    sx={{
+                      display: 'inherit',
+                      flexDirection: 'inherit',
+                      gap: 'inherit',
+                    }}
+                  >
+                    {reextracting ? (
+                      <CircularProgress size={15} sx={{ ml: 1.1 }} />
+                    ) : (
+                      <PnPReextractIconButton
+                        size="small"
+                        onClick={() => void reextract()}
+                      />
+                    )}
+                    {progressReport.pnpExtractionResult && !reextracting && (
+                      <PnPValidationIcon
+                        file={file}
+                        result={progressReport.pnpExtractionResult}
+                        size="small"
+                      />
+                    )}
+                  </Feature>
+                </>
+              )}
+            </Stack>
+          }
+          disableIcon
+          disablePreview
+          sx={{
+            width: file && 'max-content',
+          }}
           {...rest}
         />
-      </Tooltip>
+      </>
```

Comment:
Not needed.

Reply:
It was a quick disable that kept indentation. Revisit soon :tm: